### PR TITLE
Finish Bug Fixing Sprint

### PR DESCRIPTION
**Pull request by amgrubb**
This PR establishes a new stable state for `develop` based on the bug fixes in each of these commits.

### CODE REVIEW

**Comment by amgrubb** [comment]
**@amgrubb must merge this branch.**

**Comment by YesugenB** [comment]
I think for any intention if the function type is RC or CR and you touch/try to move it, you get this error

Screen Shot 2021-10-23 at 7 34 17 PM

File: leaf-ui/js/onFunctions.js [hunk 1]
```diff
@@ -275,12 +275,33 @@ graph.on("add", function (cell) {
...
         if (createdInstance.length >= 2) {
-            var lastactor = createdInstance[createdInstance.length - 2].model.attr('.name/text');
-            name = cell.attr('.name/text') + "_" + (Number.parseInt(lastactor.charAt(lastactor.length - 1)) + 1);
+            var numList = "";
+            for (let i = 2; i < createdInstance.length + 1; i++) {
+                // Gets the number from an actor's name 
+                var nameIndex = parseInt(createdInstance[createdInstance.length - i].model.attr('.name/text').split('_').pop());
+                // If name has been changed 
+                if (createdInstance[createdInstance.length - i].model.attr('.name/text').split('_').shift() == "Actor") {
+                    numList += nameIndex + " "
```
Comment (Domily11) [inline leaf-ui/js/onFunctions.js]: add ;?

**Review by Domily11 — DISMISSED** [review]
Aside from that one ";", everything looks good!

**Review by krhablutzel — DISMISSED** [review]
Other than that one semicolon, this all looks great to me as well! I'm also seeing the same error as Yesugen, but we could leave that as a future issue.

**Review by YesugenB — DISMISSED** [review]
I only managed to find that one bug (#607), everything else seems to be in working order 😎

**Review by i-m-foster — DISMISSED** [review]
It looks good to me👍

File: leaf-ui/js/onFunctions.js (hunk 1 above)
Comment (amgrubb) [reply leaf-ui/js/onFunctions.js]: I added the semicolon.

**Pull request merged**